src/main/scala/exercise2/KarplusStrong.scala

```diff
@@ -0,0 +1,37 @@
+package exercise2
+
+import exercise2.lib.StdAudio
+import scala.util.Random
+import scala.annotation.tailrec
+
+object KarplusStrong {
+  def main(args: Array[String]): Unit = {
+    loop(StdAudio.play)(whiteNoise())
+  }
+
+  def whiteNoise(frequency: Int = 440, volume: Double = 1.0): Queue[Double] = {
+    if (frequency <= 0 && volume < 0.0 && volume > 1.0)
+      throw new Throwable("Incorrect parameters")
+
+    0.to(frequency).foldLeft(new Queue[Double]()) { (acc, _) =>
+      acc
+        .enqueue(Random.between(-0.5, 0.5) * volume)
+        .asInstanceOf[Queue[Double]]
+    }
+  }
+
+  def update(queue: Queue[Double]): Queue[Double] = {
+    val decay = 0.996
+    val newQueue = queue.dequeue.get
```

Comment:
`.get` on `Option` should be avoided. would be nicer if `update` itself would return `Option[Queue[Double]]` and then in the end, when really needed, call `.get` on that `Option`. (there are ways to avoid called that get as well using IO, which we will check out in this week's lecture)